### PR TITLE
CancellationToken roslyn analyzer

### PR DESCRIPTION
**Pull request by allantargino**
Currently, CancellationToken binding is not supported (ignored) during [input conversion](https://github.com/Azure/azure-functions-durable-extension/blob/dev/src/Worker.Extensions.DurableTask/OrchestrationInputConverter.cs#L79). This analyzer will prevent users from using it inside Azure Functions Orchestrators.

### CODE REVIEW

File: src/Analyzers/Resources.resx [hunk 1]
```diff
@@ -153,4 +153,10 @@
   <data name="DurableClientBindingAnalyzerTitle" xml:space="preserve">
     <value>[DurableClient] must be used with DurableTaskClient</value>
   </data>
+  <data name="CancellationTokenOrchestrationAnalyzerMessageFormat" xml:space="preserve">
+    <value>Orchestration '{0}' is using a non-supported CancellationToken binding</value>
+  </data>
+  <data name="CancellationTokenOrchestrationAnalyzerTitle" xml:space="preserve">
+    <value>Function CancellationToken binding is not supported for OrchestrationTrigger</value>
```
Comment (bachuv) [inline src/Analyzers/Resources.resx:160]: Should we use the description that's in the in proc analyzer (pasted below) to keep it consistent? This description might be simpler.

> CancellationToken should not be used as an orchestrator function parameter.

File: src/Analyzers/Functions/Orchestration/CancellationTokenOrchestrationAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.DurableTask.Analyzers.Orchestration;
+using static Microsoft.DurableTask.Analyzers.Functions.Orchestration.CancellationTokenOrchestrationAnalyzer;
+
+namespace Microsoft.DurableTask.Analyzers.Functions.Orchestration;
+
+/// <summary>
+/// Analyzer that reports a warning when CancellationToken is used in an Azure Function Orchestration.
```
Comment (bachuv) [inline src/Analyzers/Functions/Orchestration/CancellationTokenOrchestrationAnalyzer.cs:14]:
```suggestion
/// Analyzer that reports a warning when CancellationToken is used in a Durable Functions Orchestration.
```

File: src/Analyzers/Resources.resx (hunk 1 above)
Comment (allantargino) [reply src/Analyzers/Resources.resx]: Updated :)1

File: src/Analyzers/Resources.resx [hunk 3]
```diff
@@ -153,4 +153,10 @@
   <data name="DurableClientBindingAnalyzerTitle" xml:space="preserve">
     <value>[DurableClient] must be used with DurableTaskClient</value>
   </data>
+  <data name="CancellationTokenOrchestrationAnalyzerMessageFormat" xml:space="preserve">
+    <value>Orchestration '{0}' is using a non-supported CancellationToken binding</value>
```
Comment (bachuv) [inline src/Analyzers/Resources.resx:157]:
```suggestion
    <value>Orchestration '{0}' is using a CancellationToken binding, which is not supported</value>
```

**Review by bachuv — APPROVED** [review]
Just left a small suggestion, but otherwise looks good!

**Pull request merged**